### ISSUE TITLE
applying `new_icon` to one rules changes for all  

### ISSUE THREAD

**Issue by rudevdr** [dunst-project/dunst#658]
I was looking to disable icon for all application, except one. I was looking to set `icon_position` via rule but its not available (relevent issue: https://github.com/dunst-project/dunst/issues/638).  I tried to get workaround the issue by setting `icon_position = left` in `global` and use `new_icon` in rules. Then I caught this issue.

My `dunstrc` is something like this. 

```cfg
[global]

icon_position = left

[my-app]
	appname = myapp
	summary = "*"
	timeout = 0

[fix-icon]
	summary = "*"
	new_icon = ""
```

## Expected Behaviour 

Only notifications from `myapp` should display the icon and other icons should not show icon at all.

## Actual Behaviour

No icon in all notifications. If `new_icon = ""` rule is removed, then icon in all notifications is shown.

___

It looks like setting `new_icon = ""` in one rule does not respect icon supplied to a notification in all notifications and sets them all to `null`.

**Comment by tsipinakis** [comment]
This is working as intended. Rules do not override each other and are checked independently. So in this case what happens is
* All notifications with `appname = myapp` have their timeout set to 0
* All notifications have their icons disabled

Unfortunately I don't think there are any workarounds to achieve this currently, at least without listing all appnames. For a proper solution subscribe to #645, which should allow more advanced match syntax if it's added.

**Comment by rudevdr** [comment]
> Rules do not override each other and are checked independently.

Ah, I get it now. Thanks for explaining.

As you mentioned in #645, `fnmatch` would solve this issue. I could change to:

```cfg
[my-app]
	appname = myapp
	summary = "*"
	timeout = 0

[fix-icon]
	appname = "!myapp"
	summary = "*"
	new_icon = ""
```
and this will solve this issue. 

But still it will be a workaround. Implementing solution mentioned in #638 would be perfect solution for this workaround. 

I will close it once `fnmatch` is supported. But if you think it should be close, please go ahead and close it.

**Comment by fwsmit** [comment]
Regex is implemented in #1017. It should be better than fnmatch.
